pkg/reconciler/vspheresource/vsphere.go

```diff
@@ -146,18 +146,17 @@ func (r *Reconciler) reconcileConfigMap(ctx context.Context, vms *sourcesv1alpha
 
 func (r *Reconciler) reconcileServiceAccount(ctx context.Context, vms *sourcesv1alpha1.VSphereSource) error {
 	ns := vms.Namespace
-	name := resourcenames.ServiceAccount(vms)
-
-	_, err := r.saLister.ServiceAccounts(ns).Get(name)
+	serviceAccountName := resourcenames.ServiceAccount(vms)
+	_, err := r.saLister.ServiceAccounts(ns).Get(serviceAccountName)
 	if apierrs.IsNotFound(err) {
 		sa := resources.MakeServiceAccount(ctx, vms)
 		_, err := r.kubeclient.CoreV1().ServiceAccounts(ns).Create(ctx, sa, metav1.CreateOptions{})
```

Comment:
Should we still be creating service accounts? The two options are now default or a custom one provided by the user. It might be better to error if a custom service account isn't found.